core/src/main/scala/org/apache/spark/shuffle/ObjectShuffleReader.scala

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.shuffle
+
+import java.io.InputStream
+
+import org.apache.spark.util.CompletionIterator
+import org.apache.spark.util.collection.ExternalSorter
+import org.apache.spark.{InternalAccumulator, ShuffleDependency, InterruptibleIterator, TaskContext, SparkEnv}
+import org.apache.spark.io.CompressionCodec
+import org.apache.spark.serializer.Serializer
+
+
+private[spark] class ObjectShuffleReader[K, V, C](dep: ShuffleDependency[K, V, C]) {
```

Comment:
This class is the result of splitting BlockStoreShuffleReader into two separate classes, a lower-level class which manages block fetching, and this higher-level class which manages deserialization and aggregation for the existing object / key-value-pair-based shuffle path.